### ISSUE TITLE
`unused_parens` false positive on mut reference pattern containing a mut identifier pattern

### ISSUE THREAD

**Issue by fmease** [rust-lang/rust#114911]
Given the following code

```rs
fn main() {
    let c = &mut 'c';
    let &mut (mut _x) = c;
}
```

rustc incorrectly considers the parentheses around the mut identifier pattern `mut _x` to be unused:

```
warning: unnecessary parentheses around pattern
 --> src/main.rs:3:14
  |
3 |     let &mut (mut _x) = c;
  |              ^      ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
3 -     let &mut (mut _x) = c;
3 +     let &mut mut _x = c;
  |
```

@rustbot label C-bug T-compiler A-lint

**Comment by fmease** [comment]
Oops, `let &mut mut _x = c;` is actually grammatical, it's too late :sweat_smile:
